### DISCUSSION TITLE
Sticky sections on scroll

### DISCUSSION THREAD

**Discussion by jaymehta-twilio** [twilio-labs/paste#1519] [Q&A] [answered]
Hi team,

We do have a requirement for a sticky section when it reaches to the top of the viewport. Are there any guidelines or recommendations to achieve this today ? Any plans to include in the paste library ?

<img width="1188" alt="Screen Shot 2021-06-02 at 11 33 59 AM" src="https://user-images.githubusercontent.com/52717786/120534500-c276ac00-c396-11eb-8466-76e4dde67c79.png">

**Comment by zahnster** [comment, marked as answer]
Hi @jaymehta1986,

You can use the CSS sticky position ('position: sticky') to accomplish this. 

You'll need a few things:
- `position="sticky"` on the element you want to stick to the viewport when scrolled
- `top="someSpacingVar"` on the same element (this sets how top from the viewport it should display, and you'll need to declare an actual spacing token there)

I have a simple codebox example set up for you to check out here:
https://codesandbox.io/s/hopeful-bird-v9dqm?file=/src/index.tsx

In my example, the sidebar is the sticky element. You will have to make the viewport small enough to trigger scrolling to see the effect.

Let me know if you need additional info!

**Reply by jaymehta-twilio** [reply]
@zahnster thanks much for the info. I think this should work for our use case. 🙏🏽
